src/clients/c/tb_client/signal.zig

```diff
@@ -50,15 +50,28 @@ pub const Signal = struct {
     /// Stops the current event.
     /// Further calls to "notify" will not fire the signal.
     /// Safe to call from multiple threads.
-    pub fn stop(self: *Signal) void {
+    pub fn shutdown(self: *Signal) void {
         const listening = self.listening.swap(false, .release);
         if (listening) {
             self.notify();
         }
     }
 
-    pub fn stop_requested(self: *const Signal) bool {
-        return !self.listening.load(.acquire);
+    pub fn status(self: *const Signal) enum {
+        running,
+        stop_requested,
+        shutdown,
```

Comment:
Maybe `stop_requested` and `stopped` ? To make them more clearly related to each other.

Reply:
And can we expand the doc comment for Signal to document the actual lifecycle?